sdk/js/src/relayer/__tests__/wormhole_relayer.ts

```diff
@@ -0,0 +1,690 @@
+import { afterAll, beforeEach, describe, expect, jest, test} from "@jest/globals";
+import { PublicKey } from "@solana/web3.js";
+import { ethers } from "ethers";
+import { generateRandomString, waitForRelay } from "./utils/utils";
+import {getAddressInfo} from "../consts" 
+import {getDefaultProvider} from "../main/helpers"
+import {
+    relayer,
+    ethers_contracts,
+    tryNativeToUint8Array,
+    ChainId,
+    CONTRACTS,
+    CHAIN_ID_TO_NAME
+  } from "../../../";
+  import {GovernanceEmitter, MockGuardians} from "../../../src/mock";
+
+  const env = process.env['ENV'];
+  if(!env) throw Error("No env specified: tilt or ci or testnet or mainnet");
+  const network = env == 'tilt' || env == 'ci' ? "DEVNET" : env == 'testnet' ? "TESTNET" : env == 'mainnet' ? "MAINNET" : undefined;
+  if(!network) throw Error(`Invalid env specified: ${env}`);
+
+  const sourceChainId = network == 'DEVNET' ? 2 : 6;
+  const targetChainId = network == 'DEVNET' ? 4 : 14;
+
+// Devnet Private Key
+const privateKey = process.env['WALLET_KEY'] || "4f3edf983ac636a65a842ce7c78d9aa706d3b113bce9c46f30d7d21715b23b1d"
+ 
+const guardianRPC = env == 'tilt' ? "http://localhost:7071" : env == 'ci' ? "http://guardian:7071" : env == "testnet" ? "https://wormhole-v2-testnet-api.certus.one" : env == "mainnet" ? "https://wormhole-v2-mainnet-api.certus.one" : "";
+
+const sourceAddressInfo = getAddressInfo(sourceChainId, network);
+const targetAddressInfo = getAddressInfo(targetChainId, network);
+const sourceProvider = getDefaultProvider(network, sourceChainId, env=='ci');
+const targetProvider = getDefaultProvider(network, targetChainId,  env=='ci');
+
+// signers
+const walletSource = new ethers.Wallet(privateKey, sourceProvider);
+const walletTarget = new ethers.Wallet(privateKey, targetProvider);
+
+const sourceCoreRelayerAddress = sourceAddressInfo.coreRelayerAddress;
+const sourceMockIntegrationAddress = sourceAddressInfo.mockIntegrationAddress;
+const targetCoreRelayerAddress = targetAddressInfo.coreRelayerAddress;
+const targetMockIntegrationAddress = targetAddressInfo.mockIntegrationAddress;
+
+if(!sourceCoreRelayerAddress) throw Error("No source core relayer address");
+if(!targetCoreRelayerAddress) throw Error("No source core relayer address");
+if(!sourceMockIntegrationAddress) throw Error("No source mock integration address");
+if(!targetMockIntegrationAddress) throw Error("No source mock integration address");
+
+const sourceCoreRelayer = ethers_contracts.CoreRelayer__factory.connect(
+  sourceCoreRelayerAddress,
+  walletSource
+);
+const sourceMockIntegration = ethers_contracts.MockRelayerIntegration__factory.connect(
+  sourceMockIntegrationAddress,
+  walletSource
+);
+const targetCoreRelayer = ethers_contracts.CoreRelayer__factory.connect(
+  targetCoreRelayerAddress,
+  walletTarget
+);
+const targetMockIntegration = ethers_contracts.MockRelayerIntegration__factory.connect(
+  targetMockIntegrationAddress,
+  walletTarget
+);
```

Comment:
this very much looks like you should have a `source` and a `target` object and a single function to initialize them, seeing how everything here clearly has the same "shape"

(yes it's just testing code, but we should still hold ourselves to some standard...)

Reply:
agreed, done!